unic/ucd/normal/src/decomposition_type.rs

```diff
@@ -12,228 +12,180 @@
-///
-/// * <http://www.unicode.org/reports/tr44/#Character_Decomposition_Mappings>
-#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
-#[allow(missing_docs)]
-pub enum DecompositionType {
-    Canonical,
-    Compat,
-    Circle,
-    Final,
-    Font,
-    Fraction,
-    Initial,
-    Isolated,
-    Medial,
-    Narrow,
-    Nobreak,
-    None,
-    Small,
-    Square,
-    Sub,
-    Super,
-    Vertical,
-    Wide,
-}
-
+        /// Otherwise unspecified compatibility character
+        Compat {
+            abbr => Com,
+            long => Compat,
+            human => "Unspecified",
+        }
 
-impl CharProperty for DecompositionType {
-    fn prop_abbr_name() -> &'static str {
-        "dt"
-    }
+        /// Encircled form
+        Circle {
+            abbr => Enc,
+            long => Circle,
+            human => "Encircled",
+        }
 
-    fn prop_long_name() -> &'static str {
-        "Decomposition_Type"
-    }
+        /// Final presentation form (Arabic)
+        Final {
+            abbr => Fin,
+            long => Final,
+            human => "Arabic Final",
+        }
 
-    fn prop_human_name() -> &'static str {
-        "Decomposition Type"
-    }
-}
+        /// Font variant (for example, a blackletter form)
+        Font {
+            abbr => Font,
+            long => Font,
+            human => "Font Variant",
+        }
 
+        /// Vulgar fraction form
+        Fraction {
+            abbr => Fra,
+            long => Fraction,
+            human => "Vulgar Fraction",
+        }
 
-impl PartialCharProperty for DecompositionType {
-    fn of(ch: char) -> Option<Self> {
-        Self::of(ch)
-    }
-}
+        /// Initial presentation form (Arabic)
+        Initial {
+            abbr => Init,
+            long => Initial,
+            human => "Arabic Initial",
+        }
 
+        /// Isolated presentation form (Arabic)
+        Isolated {
+            abbr => Iso,
+            long => Isolated,
+            human => "Arabic Isolated",
+        }
 
-impl EnumeratedCharProperty for DecompositionType {
-    fn all_values() -> &'static [Self] {
-        Self::all_values()
-    }
+        /// Medial presentation form (Arabic)
+        Medial {
+            abbr => Med,
+            long => Medial,
+            human => "Arabic Medial",
+        }
 
-    fn abbr_name(&self) -> &'static str {
-        self.abbr_name()
-    }
+        /// Narrow (or hankaku) compatibility character
+        Narrow {
+            abbr => Nar,
+            long => Narrow,
+            human => "Narrow",
+        }
 
-    fn long_name(&self) -> &'static str {
-        self.long_name()
-    }
+        /// No-break version of a space or hyphen
+        NoBreak {
+            abbr => Nb,
+            long => Nobreak,
+            human => "No-Break",
+        }
 
-    fn human_name(&self) -> &'static str {
-        self.human_name()
-    }
-}
+        // TODO: Is this used?
```

Comment:
No, instead of supporting `None` variant in the type, we made the type Partial, using `Option::None` for this case. So, we can drop this variant here.

Basically, the idea is to not create a new `None` type and build tooling around it while `Option::None` does the job and provides the same semantics. I'm going to update the *Unicode API* doc to explain these, as soon as we're done with the work here.